Code Review

Japhethca opened the pull request:
## What does this PR do?
It is for review of codes.

Japhethca committed:
Merge pull request #5 from JaphethCA/ft-user-upvotes-downvotes-endpoint-150574931

Feat user upvotes and downvotes endpoint 150574931

Japhethca committed:
modified package.json

Japhethca committed:
Merge branch 'ft-user-rest-api-150574931' of https://github.com/JaphethCA/More-recipe into ft-user-rest-api-150574931

Japhethca committed:
made modification to file

Japhethca committed:
mode changes for database

Japhethca committed:
Merge pull request #6 from JaphethCA/ft-user-rest-api-150574931

Ft user rest api 150574931

Japhethca committed:
added index.js file for heroku

Japhethca committed:
Merge pull request #7 from JaphethCA/ft-user-rest-api-150574931

Refact: added index.js file for heroku

Japhethca committed:
made modifications

Japhethca committed:
Merge pull request #8 from JaphethCA/ft-user-rest-api-150574931

changed files to be able to run on heroku

Japhethca committed:
modified files

Japhethca committed:
Merge pull request #9 from JaphethCA/ft-user-rest-api-150574931

chores: modified package.json

Japhethca committed:
added lines of code

Japhethca committed:
Merge pull request #10 from JaphethCA/ft-user-rest-api-150574931

added lines of code

Japhethca committed:
Update Procfile

Japhethca committed:
Update Procfile

Japhethca committed:
removed redundant files

Japhethca committed:
made modification on some files

Japhethca committed:
set up travis

Japhethca committed:
set up codeclimate

Japhethca committed:
added coveralls configuration files

Japhethca committed:
Chore: modified routes and controller files

Japhethca committed:
Chore: modified app.js

Japhethca committed:
updated route files and controllers

Japhethca committed:
Merge pull request #11 from JaphethCA/ft-user-rest-api-150574931

Feat(API) updated files for heroku 150574931

Japhethca committed:
refactor: eslinted files

Japhethca committed:
Merge pull request #12 from JaphethCA/ft-user-rest-api-150574931

refactor: eslinted files

Japhethca committed:
refactor: eslinted files

Japhethca committed:
Merge pull request #13 from JaphethCA/ft-user-rest-api-150574931

refactor: eslinted files

Japhethca committed:
update controllers

Japhethca committed:
Merge pull request #14 from JaphethCA/ft-user-rest-api-150574931

updated controllers

Japhethca committed:
modified controllsers and routes

Japhethca committed:
modied controller files

Japhethca committed:
modified controllers and routes

Japhethca committed:
Merge pull request #15 from JaphethCA/ft-user-rest-api-150574931

chores: modified controller and route files

Japhethca committed:
refactor: modified files to match Es6 syntx

Japhethca committed:
Merge pull request #16 from JaphethCA/ft-user-rest-api-150574931

refactor: modified files to match Es6 syntx

Japhethca committed:
updated travis file

Japhethca committed:
file modification

Japhethca committed:
updated readme

Japhethca committed:
Update README.md

Japhethca committed:
japhethca

Japhethca committed:
removed ingnored files

Japhethca committed:
Merge pull request #17 from JaphethCA/ft-user-rest-api-150574931

chores : refactored files

Japhethca committed:
updated travis yml

Japhethca committed:
added code climate stats to readme

Japhethca committed:
Chores: refactored code and made few changes

Japhethca committed:
testing: added new test files

Japhethca committed:
Merge pull request #19 from JaphethCA/testing

Test: testing endpoint with mocha and chai -150575175

Japhethca committed:
made changes to file

Japhethca committed:
made changes to file

Japhethca committed:
added more test files

Japhethca committed:
added coveralls configuration file

Japhethca committed:
changes configuration

Japhethca committed:
modified file

Japhethca committed:
added more configuration

Japhethca committed:
removed build files

Japhethca committed:
added database sync

Japhethca committed:
added more configurations

Japhethca committed:
removed redundant file

Japhethca committed:
made changes to file

Japhethca committed:
renamed file folder

Japhethca committed:
refactor codes

Japhethca committed:
fixed bug in voting for recipe

Japhethca committed:
refactored code

Japhethca committed:
added more configurattion

Japhethca committed:
Merge pull request #20 from JaphethCA/ft-user-rest-api-150574931

Ft user rest api 150574931

Japhethca committed:
updated coveralls configuration

Japhethca committed:
added more test scripts

Japhethca committed:
entry point for app

Japhethca committed:
added more packaged and configurations

Japhethca committed:
added support javascript and react transpilation

Japhethca committed:
added more appliacation route

Japhethca committed:
added types file to hold types for redux actions

Japhethca committed:
added files for handling login and signup request using redux thunk and action creators

Japhethca committed:
add forms for login and signup

Japhethca committed:
container for homepage components including signup and signin page containers

Japhethca committed:
added files handling authentications

Japhethca committed:
added file for setting and deleting authorization

Japhethca committed:
holds navigation components

Japhethca committed:
refactored file

Japhethca committed:
added webpack middleware

Japhethca committed:
added validation for client side

Japhethca committed:
more configuration

Japhethca committed:
build files

Japhethca committed:
commented webpackfile

Japhethca committed:
added webpack config to server folder

Japhethca committed:
re added webpack to file

Japhethca committed:
added build files

Japhethca committed:
made changes to file

Japhethca committed:
added build file

Japhethca committed:
updated package.json

Japhethca committed:
deleted uneccessay build files

Japhethca committed:
readded database migration files

Japhethca committed:
added darabase seed files

Japhethca committed:
refactored route controllers

Japhethca committed:
updated tests

Japhethca committed:
updated test

Japhethca committed:
refactored database model files

Japhethca committed:
added more files to ignore

Japhethca committed:
updated model config file

Japhethca committed:
made changes to travis config

Japhethca committed:
update travis configuration file

Japhethca committed:
add .nyc_output to ignore

Japhethca committed:
Merge pull request #21 from JaphethCA/ft-client-side-with-react-redux

feat(client) Client side implementation using react and redux-150575167

Japhethca committed:
added login and logout react pages including token authentication

Japhethca committed:
updated travis with more configuration

Japhethca committed:
Merge pull request #22 from JaphethCA/ft-client-side-with-react-redux

added login and logout react pages including token authentication

Japhethca committed:
Revert "added login and logout react pages including token authentication"

Japhethca committed:
added more component to app

Japhethca committed:
Merge pull request #23 from JaphethCA/revert-22-ft-client-side-with-react-redux

Revert "added login and logout react pages including token authentication"

Japhethca committed:
Merge pull request #24 from JaphethCA/ft-client-side-with-react-redux

added more component to app

Japhethca committed:
Update .travis.yml

Japhethca committed:
Revert "feat: added more component to app"

Japhethca committed:
updated config files

Japhethca committed:
Merge pull request #25 from JaphethCA/revert-24-ft-client-side-with-react-redux

Revert "feat: added more component to app"

Japhethca committed:
made changes to test files

Japhethca committed:
configured config.js

Japhethca committed:
Merge pull request #26 from JaphethCA/ft-client-side-with-react-redux

updated config files

Japhethca committed:
added more tests and configuartion

Japhethca committed:
Merge pull request #27 from JaphethCA/ft-client-side-with-react-redux

added more tests and configuration

Japhethca committed:
updated travis badge

Japhethca committed:
added API documentation link

Japhethca committed:
adjusted more recipe links position

Japhethca committed:
added webpack config to package.json

Japhethca committed:
added webpack configuration

Japhethca committed:
added recipe request handlers

Japhethca committed:
added recipe delete, downvote, upvote, review and update buttons

Japhethca committed:
added recipe components

Japhethca committed:
added file upoad api route/endpoint

Japhethca committed:
added flash message action handler

Japhethca committed:
updated client index files

Japhethca committed:
added scss styling files

Japhethca committed:
removed redundant files

Japhethca committed:
refacted controller files and added upload route

Japhethca committed:
added component action types

Japhethca committed:
refactored files

Commit:
made change to configuration files

Commit:
added react to js configuration

Commit:
added more route to index js file

Commit:
added more action types for the reducers

Commit:
implemented actions to handle getting recipes, reviews and userdetails

Commit:
removed redundant files

Commit:
added action controllers/dispatchers for creating recipes, upvoting and handling user profile

Commit:
added call to action button components

Commit:
added more styles customization for the components

Commit:
added user profile api route and refactored other api controllers

Commit:
added login, userrecipes and recipe update page containers

Commit:
implemented root reducer including reducers for favorites, profile, recipes, reviews and user favorites

Commit:
added myrecipes, recipes, and recipedetails conponets

Commit:
added favorites and favorites page controllers

Commit:
added styling and authentication to login page

Commit:
added page navigation and footer conponents

Commit:
added component for user to be able to add new recipe to app

Commit:
added support for reviews in app with addnew and review components for adding new review and viewing the reviews

Commit:
made style changes to sign up forms including validations

Commit:
added component for editing and viewing user profiles

Commit:
added component for showing reviews on a recipe

Commit:
component for displaying user details

Commit:
added container component for holding profile edits and view

Commit:
made chages in the layout of the component

Commit:
made changes to setauthorization and signuppage

Commit:
added more static image for recipe

Japhethca committed:
test(controllers): adds more test cases and renames test files for server side controllers

Japhethca committed:
refactor(models): refactors models files to es6

Japhethca committed:
feat (controllers):  adds more controllers for routes

Japhethca committed:
chore(server): adds model migration files, removes redundant files and adds more api routes

Japhethca committed:
chore(config): adds more configurations, removes redundant config files

Japhethca committed:
chore(travis): adds conguration for travis for run test

Japhethca committed:
chore(coveralls): configured coveralls and travis

Japhethca committed:
chore(webpack): configures webpack for client side development

Japhethca committed:
feat(client): adds component for signing in  and sign up

Japhethca committed:
feat(client): adds components for creating, viewing and updating recipe

Japhethca committed:
feat(client): adds buttons components for deleting, updating, reviews and voting recipe

Japhethca committed:
feat(UI): adds navigation component for navbars

Japhethca committed:
feat(UI): adds client side form for reviewing recipe and viewing recipe

Japhethca committed:
feat(UI): adds search capability and user profile

Japhethca committed:
style(UI): styles UI components

Japhethca committed:
fix(mergeConflict): fixes merge conflicts

Japhethca committed:
fix(travis): fixes issue with travis failing

Japhethca committed:
Merge pull request #28 from JaphethCA/ft-client-side-with-react-redux

API endpoint and configurations

Japhethca committed:
Update README.md

Japhethca committed:
chore(heroku): setup heroku

Japhethca committed:
chore(heroku): fixes issues with heroku not running

Japhethca committed:
feat(image): adds image upload preview

Japhethca committed:
feat(search): adds search form components and reducers

Japhethca committed:
refactor(codebase): renames files and folder for easy access

Japhethca committed:
refactor(server): did object destructring in server controllers

Japhethca committed:
#152843338 Refactor codebase (#30)

* chore(heroku): setup heroku

* chore(heroku): fixes issues with heroku not running

* feat(image): adds image upload preview

* feat(search): adds search form components and reducers

* refactor(codebase): renames files and folder for easy access

* refactor(server): did object destructring in server controllers

* chore(eslint): corrects hound errors

Japhethca committed:
fix(merge-conflicts): resolves merge conflicts while merging (PR-30)Refactor codebase to staging

Japhethca committed:
fix(signup): removed merge fix codes form from signup component

Japhethca committed:
feat(landingPage): adds quotes and authentication buttons

Japhethca committed:
chore(validation): validate user inputs

Japhethca committed:
chore(validation): validate user inputs

Japhethca committed:
#152676149 Implement Recipe Search (#29)

* refactor(search-UI): redesigns the search input

* feat(Email-notification): installs necessary dependencies

* feat(navigation): redesigned navigation to be more user friendly

* feat(search): adds page to show search result

* refactor(codebase): rearranges react component based on a/b specs
- adds new bundle.js and style.css files

Japhethca committed:
fix(merge): merges branch with current staging

Japhethca committed:
chore(validation): validates user input on signin, signup and other forms

Japhethca committed:
fix(Hound): fixes hounds errors

Japhethca committed:
fix(heroku): fixes issue with app not running on heroku

thechinedu commented on client/actions/requestHandlers/handleCreateRecipe.js:
```diff
@@ -0,0 +1,32 @@
+export default data => (dispatch) => {
+  if (typeof (data.image) === 'object') {
+    upload(data.image).end((err, res) => {
+      if (!err || res.ok) {
```
Do you really need both checks here? Wouldn't one of the conditions suffice?

thechinedu commented on client/actions/requestHandlers/handleCreateRecipe.js:
```diff
@@ -0,0 +1,32 @@
+      }
+    });
+  } else {
+    axios.post('/api/recipes', data).then((res) => {
```
I see this is also in the `if` block, you should have a single `if` block and this action can take place outside the conditionals (seeing as it gets run for both blocks). Something like this:

```javascript
  if (typeof(data.image) === 'object') {
     upload(data.image).end((err, res) => {
         if (!err || res.ok) {
             data.image = res.body.url;
         } else {
             Materialize.toast('Unable to load image', 4000);
         }
     });
 }
 axios.post('/api/recipes', data).then((res) => {
     dispatch(addNewRecipe(res.data.Details));
     Materialize.toast('Recipe Successfully Created!', 4000);
 }).catch(err => Materialize.toast('Recipe was not created!', 4000));
```

Japhethca replied:
I tried this method but i was getting an error. I think the reason is because is because node dispatches both the image upload and API call asynchronously.

thechinedu commented on client/actions/requestHandlers/handleFileUpload.js:
```diff
@@ -0,0 +1,25 @@
+  const paramsStr = `timestamp=${timestamp}&upload_preset=${uploadPreset}tOCLTWg6d4HQx_5BEuufdkNXU8c`;
+  const signature = sha1(paramsStr);
+  const params = {
+    api_key: '251593169873372',
```
This should be stored in an environment variable

thechinedu commented on client/actions/requestHandlers/handleLoginrequest.js:
```diff
@@ -0,0 +1,29 @@
+  };
+}
+export function handleLoginRequest(userdata) {
+  // handles loging request
```
Is this comment necessary? The function is already named aptly

thechinedu commented on client/actions/requestHandlers/handleRecipeUpdate.js:
```diff
@@ -0,0 +1,38 @@
+ * @author chidex
+ * @argument object
+ * @function
+ * it gets the filds from the form filled before submitting
```
~`filds`~ `fields`

thechinedu commented on client/actions/requestHandlers/handleRecipeUpdate.js:
```diff
@@ -0,0 +1,38 @@
+      if (!err || res) {
+        data.image = res.body.url;
+        Materialize.toast('Recipe Successfully Updated!', 4000);
+        axios.put(`/api/recipes/${data.id}`, data).then((res) => {
```
See comment above about actions that happen whether a condition is true or false

thechinedu commented on client/actions/requestHandlers/handleUserFavorites.js:
```diff
@@ -0,0 +1,40 @@
+}
+export function getFavorites(userId) {
+  return dispatch => axios.get(`/api/users/${userId}/recipes`).then((res) => {
+    const favorites = res.data.Favorites.map((favorite) => {
```
This can be shortened to:

```javascript
res.data.Favorites.map(favorite => favorite.Recipe)
```

thechinedu commented on client/actions/requestHandlers/handleUserProfile.js:
```diff
@@ -0,0 +1,46 @@
+      upload(data.photo).end((err, res) => {
+        if (!err || res.ok) {
+          data.photo = res.body.url;
+          axios.post('/api/users/profile', data).then((res) => {
```
See comment above about actions that happen whether a condition is true or not

thechinedu commented on client/components/buttons/DeleteButton.jsx:
```diff
@@ -0,0 +1,40 @@
+  constructor(props) {
+    super(props);
+
+    this.onClick = this.onClick.bind(this);
```
Add the [transform class properties plugin](https://babeljs.io/docs/plugins/transform-class-properties/) to your `.babelrc` file and you will not need to bind methods like this in the constructor. Once it's added, you can just rewrite the `onClick` method as an arrow function and it will be automatically bound to the class instance.

thechinedu commented on client/components/buttons/DownvoteButton.jsx:
```diff
@@ -0,0 +1,53 @@
+      hasUpvoted: false,
+    };
+
+    this.onClick = this.onClick.bind(this);
```
See comment above about `transform-class-properties`

thechinedu commented on client/components/buttons/FavoritesButton.jsx:
```diff
@@ -0,0 +1,72 @@
+    this.state = {
+      isFavorite: false
+    };
+    this.onClick = this.onClick.bind(this);
```
See comment above about `transform-class-properties`

thechinedu commented on client/components/buttons/ReviewButton.jsx:
```diff
@@ -0,0 +1,26 @@
+  reviews: PropTypes.arrayOf(PropTypes.object).isRequired
+};
+
+class ReviewButton extends Component {
```
This should be rewritten to be a stateless functional component seeing as it maintains no state of it's own nor does it use any of the lifecycle methods

thechinedu commented on client/components/buttons/UpdateButton.jsx:
```diff
@@ -0,0 +1,33 @@
+  recipe: PropTypes.objectOf(PropTypes.any).isRequired
+};
+
+class UpdateButton extends Component {
```
This should be rewritten to be a stateless functional component seeing as it maintains no state of it's own nor does it use any of the lifecycle methods

thechinedu commented on client/components/buttons/UpdateButton.jsx:
```diff
@@ -0,0 +1,33 @@
+  constructor(props) {
+    super(props);
+
+    this.onClick = this.onClick.bind(this);
```
See comment above about `transform-class-properties`

thechinedu commented on client/components/buttons/UpvoteButton.jsx:
```diff
@@ -0,0 +1,38 @@
+  handleUpvote: PropTypes.func.isRequired,
+};
+
+class UpvoteButton extends Component {
```
This should be rewritten to be a stateless functional component seeing as it maintains no state of it's own nor does it use any of the lifecycle methods

thechinedu commented on client/components/buttons/UpvoteButton.jsx:
```diff
@@ -0,0 +1,38 @@
+  constructor(props) {
+    super(props);
+
+    this.onClick = this.onClick.bind(this);
```
See comment above about `transform-class-properties`

thechinedu commented on client/components/favorites/Favorites.jsx:
```diff
@@ -0,0 +1,29 @@
+  favorites: PropTypes.arrayOf(PropTypes.object).isRequired,
+};
+
+class Favorites extends Component {
```
This should be rewritten to be a stateless functional component seeing as it maintains no state of it's own nor does it use any of the lifecycle methods

thechinedu commented on client/components/login/LoginForm.jsx:
```diff
@@ -0,0 +1,121 @@
+      hasErrored: false
+    };
+
+    this.onSubmit = this.onSubmit.bind(this);
```
See comment above about `transform-class-properties`

thechinedu commented on client/components/myRecipe/AddRecipe.jsx:
```diff
@@ -0,0 +1,111 @@
+      errors: '',
+      isLoading: false
+    };
+    this.onChange = this.onChange.bind(this);
```
See comment above about `transform-class-properties`

Japhethca committed:
fixes(image-url): changes the image url to point to external resource

Japhethca committed:
refactor(code): refactors client codes

Japhethca committed:
refactor(Buttons): changes button components and login components with transform class properties

noordean commented on server/app.js:
```diff
@@ -0,0 +1,64 @@
+import webpack from 'webpack';
+import webpackMiddleware from 'webpack-dev-middleware';
+import path from 'path';
+import fileupload from 'express-fileupload';
```
Put an empty line after this. modules import should be separated from file import by a line, for readability.

noordean commented on server/config/config.js:
```diff
@@ -0,0 +1,26 @@
+    dialect: 'postgres'
+  },
+  test: {
+    // username: 'chidex',
```
Remove all commented codes such as this

noordean commented on server/controllers/recipeContr.js:
```diff
@@ -0,0 +1,187 @@
+      return sequelize.query(`
+      SELECT * FROM "Recipes" AS "Recipes" ORDER BY "upvotes" DESC;`, { type: Sequelize.QueryTypes.SELECT })
+        .then(recipes => res.status(200).json({ message: 'All Recipes displayed in Descending order', List: recipes }))
+        .catch(err => res.status(400).json(err));
```
We can't be so sure this error is 400. Also, instead of sending the err object, try and provide your own custom error message

noordean commented on server/controllers/recipeContr.js:
```diff
@@ -0,0 +1,187 @@
+      return Recipes.findOne({ where: { name: req.body.name } })
+        .then((recipe) => {
+          if (recipe) {
+            return res.status(400).json({ message: 'Recipe with this name already exists' });
```
This is not 400. It's 409

noordean commented on server/controllers/recipeContr.js:
```diff
@@ -0,0 +1,187 @@
+          image: req.body.image
+        }))
+        .then((recipe) => {
+          res.status(200).json({ message: 'Recipe successfully created', Details: recipe });
```
This is 201, not 200.

noordean commented on server/controllers/recipeContr.js:
```diff
@@ -0,0 +1,187 @@
+          res.status(200).json({ message: 'Recipe successfully created', Details: recipe });
+        })
+        .catch(() => {
+          res.status(400).json({ message: 'Request was not processed' });
```
I've already made a comment about this

noordean commented on server/controllers/recipeContr.js:
```diff
@@ -0,0 +1,187 @@
+        });
+    }
+    const errors = Object.values(recipeValidator.errors.errors).map(val => val[0]);
+    res.status(403).json({ message: errors });
```
Using 403(forbidden) might not be appropriate here. Kindly read more about http status code, as it's a very important aspect of API design

noordean commented on server/controllers/recipeContr.js:
```diff
@@ -0,0 +1,187 @@
+  getRecipeById(req, res) {
+    Recipes.findOne({
+      where: {
+        id: req.params.recipeId,
```
I can't see where you are validating this 'id' before using it for querying.

noordean commented on server/controllers/recipeContr.js:
```diff
@@ -0,0 +1,187 @@
+
+  // controller for deleting recipe by recipeId
+  deleteRecipe(req, res) {
+    const thisRecipeId = req.params.recipeId;
```
A better name can be used for this variable

noordean commented on server/controllers/usersContr.js:
```diff
@@ -0,0 +1,152 @@
+
+  userProfile(req, res) {
+    if (!req.decoded.id) {
+      res.status(403).json({ message: 'You are not allowed to view this page' });
```
Consider changing this error message, as you are not rendering any view from here.

noordean commented on test/authencators.test.js:
```diff
@@ -0,0 +1,45 @@
+
+const { expect } = chai;
+
+let token = 'yJhbGciOiNiIsIncCI6IkpXVCJ9.eyJpZCI6MSwidXNlciI6Im';
```
Find a better way of doing this. You can use a `beforeEach()` to get the token

noordean commented on test/favorite.test.js:
```diff
@@ -0,0 +1,147 @@
+      .end((err, res) => {
+        expect(res).to.have.status(200);
+        expect(res).to.be.json;
+        expect(res.body).to.have.property('Favorites');
```
Testing for just the property `Favorites` might not be enough here. Assert for some of the expected data.

noordean commented on test/recipe.test.js:
```diff
@@ -0,0 +1,167 @@
+      })
+      .end((err, res) => {
+        expect(res).to.have.status(403);
+        expect(res).to.be.json;
```
It will be a good idea if you can test for the returned 'name`, `description`, `ingredient` and `direction`. The same thing applicable for other similar places

noordean reviewed (commented):
You have really done a great work, but you can still improve on some areas such as:
1. Proper error handling. Consider using middlewares in some cases to relieve your controllers
2. General naming convention. Files where a class/object(singleton) is being exported should be named in PascalCase. Kindly check `https://github.com/airbnb/javascript` to learn more on the airbnb style guide, and rename your controllers, models and some other methods and variables accordingly.
3. Correct Http Status Code. Kindly search and read more on this
Good work once again.

Japhethca committed:
updates readme file

Japhethca committed:
fix(README): updates readme page

Japhethca committed:
docs(action/reducers): adds JS docs

Japhethca committed:
test(Server): adds more test and fixes issue with previous tests

Japhethca committed:
chore(server): renamed server controllers
- documented models
- added more seeds

Japhethca committed:
style(landing): fixes stylings on the landing page

Japhethca committed:
Merge branch 'staging' of https://github.com/JaphethCA/More-recipe into staging

get latest from remote

Japhethca committed:
updates readme badges

Japhethca committed:
chore(build): updates build fies

Japhethca committed:
chore(readme): updated readme file and license file

Japhethca committed:
Add Heroku generated app.json

Japhethca committed:
#152798591 Fixes Server API Status Codes (#33)

* fix(status_codes): fixes incorrect status code responses

* fix(statusCode): changes status code on the client

Japhethca committed:
#153732760 Fix login issues (#34)

* fix(status_codes): fixes incorrect status code responses

* fix(statusCode): changes status code on the client

* fix(login/signup): fixes issues with user not being able to login into application

* feat(pagination): adds pagination to homepage and myrecipes page

Japhethca committed:
updates readme (#35)

- adds functionalities
- adds how to contribute section
- updates technologies used 
- adds coding style

Japhethca committed:
#153904840 Renames API Endpoints (#36)

* fix(status_codes): fixes incorrect status code responses

* fix(statusCode): changes status code on the client

* fix(login/signup): fixes issues with user not being able to login into application

* feat(pagination): adds pagination to homepage and myrecipes page

* chore(endpoints): renames api endpoints

Japhethca committed:
#153905172 Fixes Server API Tests (#37)

* fix(status_codes): fixes incorrect status code responses

* fix(statusCode): changes status code on the client

* fix(login/signup): fixes issues with user not being able to login into application

* feat(pagination): adds pagination to homepage and myrecipes page

* chore(endpoints): renames api endpoints

* fix(api): fixes server api tests

Japhethca committed:
feat(api-docs): documents API endpoint using swagger

* fix(status_codes): fixes incorrect status code responses

* fix(statusCode): changes status code on the client

* fix(login/signup): fixes issues with user not being able to login into application

* feat(pagination): adds pagination to homepage and myrecipes page

* chore(endpoints): renames api endpoints

* fix(api): fixes server api tests

* feat(apiDocs): documents api endpoints using swagger

* adds new build files

Japhethca committed:
chore(heroku): Configure Heroku Postbuild (#39)

* fix(status_codes): fixes incorrect status code responses

* fix(statusCode): changes status code on the client

* fix(login/signup): fixes issues with user not being able to login into application

* feat(pagination): adds pagination to homepage and myrecipes page

* chore(endpoints): renames api endpoints

* fix(api): fixes server api tests

* feat(apiDocs): documents api endpoints using swagger

* adds new build files

* chore(heroku): configures heroku postbuild script

Japhethca committed:
removed build image

Japhethca committed:
removes bundle.js

Japhethca committed:
removes bundle.js.map file

Japhethca committed:
remove style.css file

Japhethca committed:
removes style.css.map file from public

Japhethca committed:
chore(Readme): Adds API documentation link

Japhethca committed:
fix(navigation-UI) fixes navigation menu ui (#40)

* fix(status_codes): fixes incorrect status code responses

* fix(statusCode): changes status code on the client

* fix(login/signup): fixes issues with user not being able to login into application

* feat(pagination): adds pagination to homepage and myrecipes page

* chore(endpoints): renames api endpoints

* fix(api): fixes server api tests

* feat(apiDocs): documents api endpoints using swagger

* adds new build files

* chore(heroku): configures heroku postbuild script

* fix(navigation-UI): fixes issues with navigation UI styling

Japhethca committed:
fix(pages): fixes signup, landing, and login pages styling (#41)

Japhethca committed:
feat:Integrates WYSIWYG Editor (#42)

* fix(pages): fixes signup, landing, and login pages styling

* feat(WYSIWYG): integrates WYSIWYG editor

Japhethca committed:
fix(tinyme): adds wysiwyg editor dependency link

Japhethca committed:
#Adds routes force refresh (#43)

* fix(pages): fixes signup, landing, and login pages styling

* feat(WYSIWYG): integrates WYSIWYG editor

* chore: adds force refresh to routes

Japhethca committed:
Update index.html

Japhethca committed:
#Chore Codebase Restructure (#44)

* fix(pages): fixes signup, landing, and login pages styling

* feat(WYSIWYG): integrates WYSIWYG editor

* chore: adds force refresh to routes

* chore(client): codebase file restructure

Japhethca committed:
Bug/1374674893/fix client input validators (#45)

* fix(pages): fixes signup, landing, and login pages styling

* feat(WYSIWYG): integrates WYSIWYG editor

* chore: adds force refresh to routes

* chore(client): codebase file restructure

* fix(validations): fixes input validation error messages not disappearing on valid input

Japhethca committed:
Adds Js-doc Documentation (#46)

* fix(pages): fixes signup, landing, and login pages styling

* feat(WYSIWYG): integrates WYSIWYG editor

* chore: adds force refresh to routes

* chore(client): codebase file restructure

* docs: adds js-doc documentation

Japhethca committed:
Fixes app folder invalid root error

Japhethca committed:
fixes toastr url errors

Japhethca committed:
feat(Preloader): adds preloadersto show during network requests (#47)

Japhethca committed:
Fixes Cloudinary Landing page image path (#48)

* feat(Preloader): adds preloadersto show during network requests

* fix(background): fixes issue with background image not showing

Japhethca committed:
#Fix Profile Preloader (#49)

* feat(Preloader): adds preloadersto show during network requests

* fix(background): fixes issue with background image not showing

* fix(preloader): fixes issues with profile page preloader
loading when user recipes link is clicked

Japhethca committed:
adds links to technologies used

Japhethca committed:
refactor(action): adds actions to handle network request- handle network request start- handle nework request failure (#50)

Japhethca committed:
chore(client): Makes Search and Profile view Responsive (#51)

* fix(search-Profile): made search input form and profile view page responsive

* chore(server): refactors server code and renamed files to airbnb style

Japhethca committed:
chore: removes/rename duplicate files (#52)

Japhethca committed:
update swagger.json file

- extends the description on how to consume api.
- fixes typos

Japhethca committed:
#fixes issues with recipe form (#53)

* chore: removes/rename duplicate files

* fixes(form): redirects user when a recipe is created- adds tooltips to action buttons

Japhethca committed:
#Fix login errors on signup form (#54)

* chore: removes/rename duplicate files

* fixes(form): redirects user when a recipe is created- adds tooltips to action buttons

* Fix(signup): fixes issues with login server errors showing on signup form

Japhethca committed:
#fix recipe update data not showing (#55)

* chore: removes/rename duplicate files

* fixes(form): redirects user when a recipe is created- adds tooltips to action buttons

* Fix(signup): fixes issues with login server errors showing on signup form

* fix(update-form): fix issues with update data not showing on update form

Japhethca committed:
Chore(styles): updates singleRecipePage font size

Japhethca committed:
Fix(email): fixes issues with auth errors with cased email (#56)
